### PR TITLE
Preflight

### LINKED ISSUE #808: Pre-flight Checklist for Chainguard Enforce Installation
**What topic are you requesting a resource about?**
* **Chainguard product**
* Open source related
* Conceptual security related
* Other (please describe)

**Proposed title:**
Pre-flight Checklist for Chainguard Enforce Installation

**Description:**
Rundown of what you need to have in place for a successful Enforce installation
Full description TBD

### PR DESCRIPTION
**Pull request by SharpRake**
## Type of change

This adds a preflight checklist to the Installation section for Chainguard Enforce.

### What should this PR do?

Resolves #808 

### How should this PR be tested?

Creating this as a draft to start

### CODE REVIEW

File: content/chainguard/chainguard-enforce/installation/preflight-checklist.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+
+## General Installation Checklist
+
+The following is a list of prerequisites required across all the different installation methods available for Chainguard Enforce. 
+
+- [x] Every installation method requires you to have access to Chainguard Enforce. Sign up using [this inquiry form](https://www.chainguard.dev/contact?utm_source=docs) if you don't already have access.
+- [X] Most installation methods require you to have `chainctl` installed. You can achieve this by following our [How to Install `chainctl` tutorial](https://edu.chainguard.dev/chainguard/chainguard-enforce/how-to-install-chainctl/).
+- [x] Some installation methods require you to have a special tool like [Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli#install-terraform) or [Helm](https://helm.sh/docs/intro/install/) installed on your computer.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-enforce/installation/preflight-checklist.md:30]: Maybe?

```suggestion
- [x] Some installation methods require you to have a special tool like [Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli#install-terraform) or [Helm](https://helm.sh/docs/intro/install/) installed on your computer or system. 
```

File: content/chainguard/chainguard-enforce/installation/preflight-checklist.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+This document is intended to serve as a quick reference outlining the various components you need in order to install Chainguard Enforce under each currently-available method. It starts with a general list of prerequisites across the various installation methods. It then provides a brief overview of every installation method and highlights the prerequisites needed for each.
+
+Note that, in addition to the prerequisites listed here, there are [additional requirements for the Chainguard Enforce Agent](/chainguard/chainguard-enforce/reference/installation-requirements/) as well as [network requirements](/chainguard/chainguard-enforce/reference/network-requirements/) one should consider when installing Chainguard Enforce.
+
+
+## General Installation Checklist
+
+The following is a list of prerequisites required across all the different installation methods available for Chainguard Enforce. 
+
+- [x] Every installation method requires you to have access to Chainguard Enforce. Sign up using [this inquiry form](https://www.chainguard.dev/contact?utm_source=docs) if you don't already have access.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-enforce/installation/preflight-checklist.md:28]: What if you removed the hyphens here and just had the checkboxes?

File: content/chainguard/chainguard-enforce/installation/preflight-checklist.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+A software agent is a program that performs certain actions on behalf of a user. A software agent has the authority to make decisions on its own and doesn’t necessarily require user input. Accordingly, the Chainguard Enforce Agent is able to analyze a given cluster and enforce any policies you’ve applied to the cluster without manual human oversight.
+
+Instead of installing the Agent directly on your cluster, you can allow a remote Agent — managed by Chainguard — to access your cloud account resources on your behalf. A SaaS offering of Enforce, Agentless connections can be useful since they don’t consume any cluster resources. Because they’re managed by Chainguard, issues can often be addressed more quickly than clusters using agentful connections. Additionally, Agentless connections allow for read-only access to your clusters, which some users may find desirable. Be aware that Agentless connections are limited to public GKE and EKS clusters and Google Cloud Run, AWS ECS, and AWS AppRunner projects.
+
+The following table outlines some of the key differences between the Agent and Agentless connections.
+
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-enforce/installation/preflight-checklist.md:48]: What do you think of making the left column bold as they're labels, and title casing `Yes` and `No` for consistency?

File: content/chainguard/chainguard-enforce/installation/preflight-checklist.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+## Installing the Chainguard Enforce Agent
+
+When installing the Chainguard Enforce Agent, you can do so either interactively or declaratively. 
+
+### Installing the Agent interactively
+
+To install Chainguard Enforce interactively, you need to have been granted access and to have `chainctl` installed. You must also know whether the resources where you're installing Enforce are public or private. Private resources require the `--private` option when installing with `chainctl`.
+
+You need the following to [**install Chainguard Enforce declaratively with `chainctl`**](/chainguard/chainguard-enforce/installation/alternative-installation-methods/#install-with-chainctl).
+- [ ] [Access to Chainguard Enforce](https://www.chainguard.dev/contact?utm_source=docs).
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-enforce/installation/preflight-checklist.md:69]: If you think it looks good above, I would remove the hyphens / bullet points here too

File: content/chainguard/chainguard-enforce/installation/preflight-checklist.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+- [ ] [`chainctl` installed on your local machine](https://edu.chainguard.dev/chainguard/chainguard-enforce/how-to-install-chainctl/).
+- [ ] You should also know whether the resources you're installing it onto are public or private.
+
+
+### Installing the Agent declaratively
+
+There are currently two methods available for installing the Chainguard Enforce Agent declaratively: using HELM or using YAML. As with installing interactively, you must also take additional steps if you are using a declarative method to [install the Chainguard Enforce Agent on a private cluster](/chainguard/chainguard-enforce/installation/alternative-installation-methods/#additional-authentication-for-private-clusters).
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-enforce/installation/preflight-checklist.md:76]: Official docs have it as "Helm"

```suggestion
There are currently two methods available for installing the Chainguard Enforce Agent declaratively: using Helm or using YAML. As with installing interactively, you must also take additional steps if you are using a declarative method to [install the Chainguard Enforce Agent on a private cluster](/chainguard/chainguard-enforce/installation/alternative-installation-methods/#additional-authentication-for-private-clusters). 
```

File: content/chainguard/chainguard-enforce/installation/preflight-checklist.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+- [ ] [`chainctl` installed on your local machine](https://edu.chainguard.dev/chainguard/chainguard-enforce/how-to-install-chainctl/).
+- [ ] You should also know whether the resources you're installing it onto are public or private.
+
+
+### Installing the Agent declaratively
+
+There are currently two methods available for installing the Chainguard Enforce Agent declaratively: using HELM or using YAML. As with installing interactively, you must also take additional steps if you are using a declarative method to [install the Chainguard Enforce Agent on a private cluster](/chainguard/chainguard-enforce/installation/alternative-installation-methods/#additional-authentication-for-private-clusters). 
+
+To install Chainguard Enforce declaratively [with YAML](/chainguard/chainguard-enforce/installation/alternative-installation-methods/#declarative-option-1--install-with-yaml) or [with Helm](/chainguard/chainguard-enforce/installation/alternative-installation-methods/#declarative-option-2--install-with-a-helm-chart), you must have the following.
+- [ ] [Access to Chainguard Enforce](https://www.chainguard.dev/contact?utm_source=docs).
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-enforce/installation/preflight-checklist.md:79]: If you remove hyphens above, remove here too

File: content/chainguard/chainguard-enforce/installation/preflight-checklist.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+
+## Installing Chainguard Enforce in Agentless mode
+
+Instead of installing the Chainguard Enforce Agent directly on your cluster, you can allow a remote Agent — managed by Chainguard — to access your cloud account resources on your behalf. Known as “Agentless connections” these can be useful since they don’t consume any cluster resources and, because they’re managed by Chainguard, issues can often be addressed more quickly than clusters using agentful connections.
+
+There are two main approaches you can take to install Agentless Chainguard Enforce. You can use [Chainguard Enforce's Discovery feature](/chainguard/chainguard-enforce/chainguard-enforce-discovery-onboarding/) (either through the Console, with `chainctl`, or with Terraform) to discover your cloud resources and install Chainguard Enforce in Agentless mode onto them. You can alternatively set up a cloud account association between Chainguard Enforce and your GCP or AWS account and then install Enforce in Agentless mode manually.
+
+### Installing Agentless with Discovery using the Chainguard Enforce Console
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-enforce/installation/preflight-checklist.md:91]: Maybe? Just to make less verbose

```suggestion
### Installing Agentless with Discovery using the Console
```

**Review by ltagliaferri — APPROVED** [review]
The new organization is great and I like your table. Maybe the table can be used elsewhere also?

File: content/chainguard/chainguard-enforce/installation/preflight-checklist.md (hunk 2 above)
Comment (SharpRake) [reply content/chainguard/chainguard-enforce/installation/preflight-checklist.md:28]: I didn't think this was possible but I just learned how to do inline CSS to hide the bullets on this page (and this page alone).

File: content/chainguard/chainguard-enforce/installation/preflight-checklist.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+tags: ["Enforce", "chainctl", "Product"]
+images: []
+menu:
+  docs:
+    parent: "installation"
+toc: true
+weight: 005
+---
+
+There are a few things you need to have in place in order to install Chainguard Enforce. However, there's more than one offering of Chainguard Enforce and there are multiple different ways to install them. Because each installation method has its own set of prerequisites, it can be confusing to know exactly what you need to have in place before you install Chainguard Enforce.
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/chainguard-enforce/installation/preflight-checklist.md:17]: much better! thanks!

**Pull request merged**